src/librustc/mir/interpret/allocation.rs

```diff
@@ -22,6 +22,42 @@ pub enum InboundsCheck {
     MaybeDead,
 }
 
+/// Used by `check_in_alloc` to indicate context of check
+#[derive(Debug, Copy, Clone, RustcEncodable, RustcDecodable, HashStable)]
+pub enum CheckInAllocMsg {
+    ReadCStr,
+    CheckBytes,
+    WriteBytes,
+    WriteRepeat,
+    ReadScalar,
+    WriteScalar,
+    SlicePatCoveredByConst,
+    ReadDiscriminant,
+    CheckAlign,
+    ReadBytes,
+    CopyRepeatedly,
+    CheckBounds,
+}
```

Comment:
I think it is not necessary to go quite as fine-grained here (and that would make adding new usage sites very painful). For now I'd roughly distinguish

* memory access (read/write, make them separate if you want but I think it is fine to treat them together)
* testing if a pointer is NULL
* testing if a pointer is out-of-bounds
* pointer arithmetic (used only in Miri)

Reply:
fixed

Reply:
@RalfJung ?

Reply:
I will look at this once I have time. I understand that you want to see your contribution accepted as quickly as possible, but remember we all do this in our free time. :)